tests/unit/test_observer.py

```diff
@@ -11,30 +11,43 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-from typing import List
-
-import numpy.testing as npt
 import numpy as np
 import pytest
 import tensorflow as tf
 
+from trieste.data import Dataset
 from trieste.observer import map_is_finite, filter_finite
 
+from tests.util.misc import ShapeLike, assert_datasets_allclose
+
 
-def _nan_at_origin(t: tf.Tensor) -> tf.Tensor:
+def _sum_with_nan_at_origin(t: tf.Tensor) -> tf.Tensor:
     is_at_origin = tf.reduce_all(t == [[0., 0.]], axis=-1, keepdims=True)
```

Comment:
A short comment saying what this method does may be helpful here. The function name alone didn't quite do it for me.

Reply:
will do